src/crimson/osd/pg_backend.cc

```diff
@@ -591,13 +591,14 @@ seastar::future<> PGBackend::omap_get_vals(
         const auto& [key, value] = *iter;
         if (key.substr(0, filter_prefix.size()) != filter_prefix) {
           break;
-        } else if (num++ >= max_return ||
+        } else if (num >= max_return ||
             result.length() >= local_conf()->osd_max_omap_bytes_per_request) {
           truncated = true;
           break;
         }
         encode(key, result);
         encode(value, result);
+        ++num;
```

Comment:
i think this change was included by 4d93b817ab9a3edcac193ab5699cb9ffb05478e9 already.

Reply:
No, this bug actually exists in two functions, the first function (omap_get_keys) was fixed by @rzarzynski 's PR. After debugging we found that it exists in omap_get_vals as well, and Radek suggested that I add the fix as part of this PR.

Reply:
ahh, they looked identical! thanks!